### PR TITLE
Adding breaking change entry for retrievers

### PR DESCRIPTION
**Pull request by pmpailis**
Updating exception thrown when cluster_feature is not present for `rrf` and `text_similarity_reranker` and adding breaking change entry for the work taken place in 8.16  (https://github.com/elastic/elasticsearch/pull/112648)

### CODE REVIEW

File: docs/changelog/115399.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  title: Reworking RRF retriever to be evaluated during rewrite phase
+  area: REST API
+  details: |-
+    In this release (8.16), we have introduced major changes to the retrievers framework and how they can be evaluated,
+    focusing mainly on compound retrievers like `rrf` and `text_similarity_reranker`, which allowed us to support
+    additional search features like collapsing, explaining, aggregations, and highlighting.
+    To ensure consistency and avoid unexpected failures, we have introduced a new cluster_feature that
+    all nodes must satisfy, in order for the `rrf` and `text_similarity_reranker` retriever queries to be evaluated.
+    This guarantees that all nodes are able to take advantage of the new framework and provide compatible responses.
+    As part of the rework, we have also removed the `_rank` property from the responses of an `rrf` retriever.
```
[the remark this answers is not in the document]
Comment (pmpailis) [reply docs/changelog/115399.yaml]: Thanks Ben, that's fair! Reworded it slightly, lmtk what you think :)

File: docs/changelog/115399.yaml [hunk 2]
```diff
@@ -10,9 +10,8 @@ breaking:
     In this release (8.16), we have introduced major changes to the retrievers framework and how they can be evaluated,
     focusing mainly on compound retrievers like `rrf` and `text_similarity_reranker`, which allowed us to support
     additional search features like collapsing, explaining, aggregations, and highlighting.
-    To ensure consistency and avoid unexpected failures, we have introduced a new cluster_feature that
-    all nodes must satisfy, in order for the `rrf` and `text_similarity_reranker` retriever queries to be evaluated.
-    This guarantees that all nodes are able to take advantage of the new framework and provide compatible responses.
+    To ensure consistency and avoid unexpected failures, `rrf` and `text_similarity_reranker` retriever queries are no longer compatible
+    with previous releases ( <= 8.15).
```
Comment (benwtrent) [inline docs/changelog/115399.yaml:14]: YIKES, this implies that `rrf` and `text_similarity_reranker` no longer works! Please make sure you are very clear that they only break in a mixed cluster scenario, not that they stop working in 8.16 :)

Comment (pmpailis) [reply docs/changelog/115399.yaml:14]: hahah that's definitely not what I was going for 😅 had the mixed cluster description in the `impact` section below,
so I thought to not include it here as well. 

WDYT about the following? 
```
In this release (8.16), we have introduced major changes to the retrievers framework 
and how they can be evaluated, focusing mainly on compound retrievers 
like `rrf` and `text_similarity_reranker`, which allowed us to support full 
composability (i.e. any retriever can be nested under any compound retriever), 
as well as supporting additional search features like collapsing, explaining, 
aggregations, and highlighting.

To ensure consistency, and given that this rework is not available until 8.16, 
`rrf` and `text_similarity_reranker`  retriever queries would now  
throw an exception in a mixed cluster scenario, where there are nodes 
both in current or later (i.e. >= 8.16) and previous ( <= 8.15) versions.

As part of the rework, we have also removed the `_rank` property from 
the responses of an `rrf` retriever.
```

**Comment by pmpailis** [comment]
@elasticmachine update branch

**Review by benwtrent — APPROVED** [review]

**Comment by samxbr** [comment]
Friendly reminder that this PR seems to be a breaking change for 9.0, but is missing from the 9.0 release note. We may want to add an entry to the breaking change section of [9.0 release note](https://github.com/elastic/elasticsearch/blob/cf536f894756f25c7ae874b4211a6b00372b39a1/docs/reference/release-notes/9.0.0.asciidoc#L8).

**Pull request merged**